examples/pipeline_wavernn/wavernn.py

```diff
@@ -0,0 +1,586 @@
+        loss_item = loss.item()
+        sums["loss"] += loss_item
+        metric("loss", loss_item)
+
+        optimizer.zero_grad()
+        loss.backward()
+
+        if args.clip_norm > 0:
+            gradient = torch.nn.utils.clip_grad_norm_(
+                model.parameters(), args.clip_norm
+            )
+            sums["gradient"] += gradient
+            metric("gradient", gradient.item())
+
+        optimizer.step()
+
+        metric("iteration", sums["iteration"])
+        metric("time", time() - start2)
+        metric.print()
+        sums["iteration"] += 1
+
+        if SIGNAL_RECEIVED:
+            return
+
+    avg_loss = sums["loss"] / len(data_loader)
+
+    metric = MetricLogger("train_epoch")
+    metric("epoch", epoch)
+    metric("loss", avg_loss)
+    if "gradient" in sums:
+        metric("gradient", sums["gradient"] / len(data_loader))
+    metric("lr", scheduler.get_last_lr()[0])
+    metric("time", time() - start1)
+    metric.print()
+
+    scheduler.step()
+
+
+def evaluate(model, mode, criterion, data_loader, device, epoch):
+
+    with torch.no_grad():
+
+        model.eval()
+        sums = defaultdict(lambda: 0.0)
+        start = time()
+
+        for waveform, specgram, target in bg_iterator(data_loader, maxsize=2):
+
+            waveform = waveform.to(device)
+            specgram = specgram.to(device)
+            target = target.to(device)
+
+            output = model(waveform, specgram)
+
+            if mode == 'waveform':
+                output = output.transpose(1, 2)
+                target = target.long()
+
+            elif mode == 'mol':
+                target = target.unsqueeze(-1)
+
+            else:
+                raise ValueError(
+                    f"Expected mode: `waveform` or `mol`, but found {mode}"
+                )
+
+            loss = criterion(output, target)
+            sums["loss"] += loss.item()
+
+            if SIGNAL_RECEIVED:
+                break
+
+        avg_loss = sums["loss"] / len(data_loader)
+
+        metric = MetricLogger("validation")
+        metric("epoch", epoch)
+        metric("loss", avg_loss)
+        metric("time", time() - start)
+        metric.print()
+
+        return avg_loss
+
+
+def main(args):
+
+    devices = ["cuda" if torch.cuda.is_available() else "cpu"]
+
+    logging.info("Start time: {}".format(str(datetime.now())))
+
+    # Empty CUDA cache
+    torch.cuda.empty_cache()
+
+    # Install signal handler
+    signal.signal(signal.SIGUSR1, lambda a, b: signal_handler(a, b))
+
+    # use torchaudio transform to get waveform and specgram
+
+#    melkwargs = {
+#        "n_fft": 2048,
+#        "n_mels": args.n_freq,
+#        "hop_length": args.hop_length,
+#        "f_min": args.f_min,
+#        "win_length": args.win_length
+#    }
+
+#    transforms = torch.nn.Sequential(
+#        torchaudio.transforms.MelSpectrogram(
+#            sample_rate=args.sample_rate, **melkwargs
+#        ),
+#    )
```

Comment:
intentional?

Reply:
please add a comment why this is needed.

nit: align comment with rest of code